src/coreclr/jit/gentree.cpp

```diff
@@ -1015,6 +1015,78 @@ bool GenTreeCall::IsPure(Compiler* compiler) const
            compiler->s_helperCallProperties.IsPure(compiler->eeGetHelperNum(gtCallMethHnd));
 }
 
+//------------------------------------------------------------------------------
+// getArrayLengthFromAllocation: Return the array length for an array allocation
+//                               helper call.
+//
+// Arguments:
+//    tree           - The array allocation helper call.
+//    block          - tree's basic block.
+//
+// Return Value:
+//    Return the array length node.
+
+GenTree* Compiler::getArrayLengthFromAllocation(GenTree* tree DEBUGARG(BasicBlock* block))
+{
+    assert(tree != nullptr);
+
+    GenTree* arrayLength = nullptr;
+
+    if (tree->OperGet() == GT_CALL)
+    {
+        GenTreeCall* call = tree->AsCall();
+        if (call->gtCallLateArgs == nullptr)
+        {
+            // Currently this function is only profitable during the late stages
+            // so we do not bother handling non-late args here
+            return nullptr;
+        }
```

Comment:
```suggestion
        if (call->fgArgInfo == nullptr)
        {
            // Currently this function is only profitable during the late stages
            // so we avoid complicating below code to access the early args.
            return nullptr;
        }
```
There isn't a 1:1 correspondence between `gtCallLateArgs` and args, so the previous check is a sort of strange way to check this (it would work fine, but only because all ABIs we support pass the first two arguments in registers).

Reply:
Thanks!